### ISSUE TITLE
[GS] `download_to` with non-existent files don't raise any exception

### ISSUE THREAD

**Issue by aosh-ab7e** [drivendataorg/cloudpathlib#430]
Version: 0.18.1 

Code Example:
```py
from cloudpathlib import CloudPath

# download a non-existent file, in an existing bucket
path = CloudPath("gs://some_existent_bucket/non_existent.txt")
path.download_to(".") # should raise an exception, as the file doesn't exist

# NOTE: trying to download from a non-existing bucket will properly fail
path = CloudPath("gs://non_existent_bucket/foo.txt")
path.download_to(".") # will raise an exception
```

**Comment by pjbull** [comment]
Confirmed this issue repros. The root cause is that [we don't do an existence check before trying to download](https://github.com/drivendataorg/cloudpathlib/blob/fabb0e43d677e518b5918db9db8d41c56d477071/cloudpathlib/cloudpath.py#L893-L909), and we assume if `is_file` is False that it is a directory.

For now, the workaround is for users to add a `.exists()` check before calling `.download_to()`.

We can also consider adding this check to our implementation of `download_to`, the downside is that adds another network call so can slow things down. Ideally, we add this at the same time we implement metadata caching so that we can do this in one network call.

**Comment by aosh-ab7e** [comment]
I thought you do an existence check:
https://github.com/drivendataorg/cloudpathlib/blob/fabb0e43d677e518b5918db9db8d41c56d477071/cloudpathlib/gs/gsclient.py#L126-L128
(For `gs://` paths, I believe the method above will be called)

AFAIK `get_blob` DO checks if the file exists on the bucket and returns None if the file is absent.

**Comment by pjbull** [comment]
Unfortunately, `_download_file()` never gets called because `is_file()` False (following `pathlib` behavior for files that don't exist). We would have to add the `exists` call as I mentioned to do this in a cloud agnostic way.

**Comment by pjbull** [comment]
@aosh-ab7e This is released now in [v0.19.0](https://pypi.org/project/cloudpathlib/) if you want to upgrade and test your use case.
